library/src/main/java/com/okta/appauth/android/OktaAppAuth.java

```diff
@@ -569,20 +583,35 @@ private AuthorizationService createAuthorizationService(Context context) {
 
     @WorkerThread
     private void doAuth(PendingIntent completionIntent, PendingIntent cancelIntent) {
-        try {
-            mAuthIntentLatch.await();
-        } catch (InterruptedException ex) {
-            Log.w(TAG, "Interrupted while waiting for auth intent");
-        }
-
         Log.d(TAG, "Starting authorization flow");
-        createAuthorizationServiceIfNeeded().performAuthorizationRequest(
-                mAuthRequest.get(),
+        AuthorizationRequest request = mAuthRequest.get();
+        warmUpBrowser(request.toUri());
+        mAuthService.performAuthorizationRequest(
+                request,
                 completionIntent,
                 cancelIntent,
                 mAuthIntent.get());
     }
 
+    private void doEndSession(PendingIntent completionIntent, PendingIntent cancelIntent) {
+        Log.d(TAG, "Starting end session flow");
+
+        EndSessionRequest request = new EndSessionRequest(
+                mAuthStateManager.getCurrent().getAuthorizationServiceConfiguration(),
+                mAuthStateManager.getCurrent().getIdToken(),
+                mConfiguration.getmEndSessionRedirectUri());
```

Comment:
Do we want to support the [`state` param](https://developer.okta.com/docs/api/resources/oidc#request-parameters-4) here too?